R/utils.R

```diff
@@ -117,9 +124,9 @@ parse_labels <- function(string) {
     # If this is a misattributed data field or blank, throw warning in
     # multi_choice_to_labels
     if (length(out[[1]]) > 1 && !all(is.na(out[[1]]))) {
-      stop(paste0("Cannot parse the select_choices_or_calculations field from
+      cli_abort("Cannot parse the select_choices_or_calculations field from
                   REDCap metadata. This may happen if there is a pipe character
-                  `|` inside the label: ", string))
+                  `|` inside the label: {string}")
```

Comment:
This is the same code as above. I would de-duplicate.

Reply:
These are similar errors but actually have different root causes, I'm updating the language for the first error to reflect this and assigning them both an equivalent class as well as an individual one:

```r
cli_abort(
        "Cannot parse the select_choices_or_calculations field from
        REDCap metadata. This may happen if there is a comma separator missing
        inside the label: {string}",
        class = c("label_parse_error", "comma_parse_error", "REDCapTidieR_cond")
      )
```

```r
      cli_abort(
        "Cannot parse the select_choices_or_calculations field from
        REDCap metadata. This may happen if there is a pipe character
        `|` inside the label: {string}",
        class = c("label_parse_error", "matrix_parse_error", "REDCapTidieR_cond")
      )
```

The latter one actually needed a minor update in the logic as well.